src/Features/Core/Portable/CommentSelection/ICommentSelectionService.cs

```diff
@@ -0,0 +1,16 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System.Collections.Immutable;
+using System.Threading;
+using Microsoft.CodeAnalysis.Host;
+using Microsoft.CodeAnalysis.Text;
+
+namespace Microsoft.CodeAnalysis.CommentSelection
+{
+    internal interface ICommentSelectionService : ILanguageService
+    {
+        CommentSelectionInfo GetInfo(SourceText sourceText, TextSpan textSpan);
```

Comment:
(actually, in that last case, you could state that you don't support either form of comment and Roslyn will then do nothing...)